### PR TITLE
feat: functionality

### PR DESCRIPTION
**Pull request by Vishruta-Patil**
Functionality completed 
- Fetching questions based on categories
- Routing based on question number and category
- Result

### CODE REVIEW

File: src/components/quiz/QuizCard.jsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        </p>
+        <p className="quit-option">Timer</p>
+      </div>
+      <div className="quiz-question font-md black-color">{item.question}</div>
+      <div className="quiz-option font-sm" >
+        {item.options.map((option, index) => (
+          <div
+            className={activeOption ? "option-unit active flex" : "option-unit flex"}
+            key={index}
+            // onClick={(e) => { e.target.style.color = "#0097a7"; answerHandler(option, index)}}
```
Comment (prathmeshjagtap) [inline src/components/quiz/QuizCard.jsx:65]: you can remove commented code if not needed

**Review by prathmeshjagtap — APPROVED** [review]
Code is Clean
Learned window scroll to top feature

File: src/components/Router.js [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
+import { Result } from "../pages/result/result";
 
 export const Router = () => {
     return (
         <Routes>
             <Route path="/" element={<Home />}/>
             <Route path="/instructions" element={<Instructions />}/>
-            <Route path="/instructions" element={<Instructions />}/>
+            <Route path="/quiz/:question" element={<Quiz />}/> 
+            <Route path="/result" element={<Result />}/>
```
Comment (shivanigangadharan) [inline src/components/Router.js:13]: Can remove the `/` as it's not reqd in react v6.

File: src/pages/instructionPage/index.js [hunk 3]
```diff
@@ -1,24 +1,31 @@
...
+import { SET_QUESTION_NUMBER } from "../../reducer/constant"
 import "./instruction.css"
 
 export const Instructions = () => {
+    const {quizState, quizDispatch} = useQuiz()
+    const {category, questionNo} = quizState
+    
     useEffect(() => {
         window.scrollTo(0,0)
+        // quizDispatch({type: SET_QUESTION_NUMBER})
```
Comment (shivanigangadharan) [inline src/pages/instructionPage/index.js:15]: can remove comment if not reqd

File: src/pages/quiz/index.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    const {category, questionNo} = quizState
+
+    const {question} = useParams()
+    console.log(question)
+    
+  return (
+    <div>
+      {/* {quizData[category].map((item, index) => ( */}
+        <QuizCard item={quizData[category][questionNo-1]} />
+      {/* ))} */}
```
Comment (shivanigangadharan) [inline src/pages/quiz/index.js:18]: if not reqd can remove comments here too

**Review by shivanigangadharan — APPROVED** [review]
Code looks good, just a suggestion to also add a deploy preview/gif for better understanding of UI.

**Pull request merged**